astroquery/gemini/core.py

```diff
@@ -0,0 +1,480 @@
+                'GMOS',
+                'GMOS-N',
+                'GMOS-S',
+                'GNIRS',
+                'GRACES',
+                'NIRI',
+                'NIFS',
+                'GSAOI',
+                'F2',
+                'GPI',
+                'NICI',
+                'MICHELLE',
+                'TRECS',
+                'BHROS',
+                'HRWFS',
+                'OSCIR',
+                'FLAMINGOS',
+                'HOKUPAA+QUIRC',
+                'PHOENIX',
+                'TEXES',
+                'ABU',
+                'CIRPASS'
+        observation_class : str, optional
+            Specifies the class of observations to search for.  Valid values are:
+                'science',
+                'acq',
+                'progCal',
+                'dayCal',
+                'partnerCal',
+                'acqCal'
+        observation_type : str, optional
+            Search for a particular type of observation.  Valid values are:
+                'OBJECT',
+                'BIAS',
+                'DARK',
+                'FLAT',
+                'ARC',
+                'PINHOLE',
+                'RONCHI',
+                'CAL',
+                'FRINGE',
+                'MASK'
+        mode : str, optional
+            The mode of the observation.  Valid values are:
+                'imaging',
+                'spectroscopy',
+                'LS',
+                'MOS',
+                'IFS'
+        adaptive_optics : str, optional
+            Specify the presence of adaptive optics.  Valid values are:
+                'NOTAO',
+                'AO',
+                'NGS',
+                'LGS'
+        program_text : str, optional
+            Specify text in the information about the program.  This is free form text.
+        objectname : str, optional
+            Give the name of the target.
+        raw_reduced : str, optional
+            Indicate the raw or reduced status of the observations to search for.  Valid values are:
+                'RAW',
+                'PREPARED',
+                'PROCESSED_BIAS',
+                'PROCESSED_FLAT',
+                'PROCESSED_FRINGE',
+                'PROCESSED_ARC'
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+
+        # This could be refactored to rely on query_raw, but I am not convinced
+        # that is a good long-term approach.  I prefer the clarity of keeping
+        # this as an independent implementation.
+        if radius is not None:
+            if isinstance(radius, (int, float)):
+                radius = radius * units.deg
+            radius = astropy.coordinates.Angle(radius)
+
+        url = "%s/jsonsummary/notengineering/NotFail" % self.server
+        if coordinates is not None:
+            coordinates = commons.parse_coordinates(coordinates)
+            url = "%s/ra=%f/dec=%f" % (url, coordinates.ra.deg, coordinates.dec.deg)
```

Comment:
Instead of building up a URL like this, you could build up a dictionary of keyword/value pairs and format them all later.  But this looks like a RESTful API, maybe?  Does the order of arguments matter?  If not, I suggest a dict refactor.